### PR TITLE
Prepare mkcloud for running 7-8 upgrade

### PR DESCRIPTION
**Pull request by jsuchome**

### CODE REVIEW

File: scripts/mkcloud [hunk 1]
```diff
@@ -955,16 +955,19 @@ function onhost_reset_admin
...
-        onadmin upgrade_ses_to_4
+        cloudsource=$upgrade_cloudsource onadmin prepare_cloudupgrade_nodes_repos
+        if [[ $cloudsource = develcloud6 ]]; then
+            onadmin upgrade_ses_to_4
+        fi
         onadmin zypper_patch_all
         onadmin upgrade_prechecks
-        if [[ $cloudsource = develcloud6 ]] || [[ $upgrade_cloudsource = develcloud7 ]]; then
+        if [[ $cloudsource = develcloud6 ]] || [[ $upgrade_cloudsource = develcloud7 ]] || \
+           [[ $cloudsource = develcloud7 ]] || [[ $upgrade_cloudsource = develcloud8 ]]; then
```
Comment (bmwiedemann) [inline scripts/mkcloud]: how about simplifying the condition as 
```bash
[[ $cloudsource =~ devel ]] || [[ $upgrade_cloudsource =~ devel ]]
```

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -4676,8 +4676,14 @@ function onadmin_prepare_cloudupgrade_admin_repos_6_to_7
     safely createrepo $tftpboot_repos_dir/PTF
 
     # change system repositories to SP2
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: with the change below this comment becomes a bit misleading - maybe remove it as the code is rather clear - or have two copies within the 'if's

File: scripts/mkcloud (hunk 1 above)
Comment (jsuchome) [reply scripts/mkcloud]: Yep, good one

Comment (bmwiedemann) [reply scripts/mkcloud]: the (less readable?) variant with further reduced duplication would be
```bash
[[ $cloudsource$upgrade_cloudsource =~ devel ]]
```

File: scripts/mkcloud [hunk 3]
```diff
@@ -955,16 +955,18 @@ function onhost_reset_admin
 function crowbarupgrade_5plus
 {
     if iscloudver 6plus ; then
-        cloudsource=$upgrade_cloudsource onadmin prepare_cloudupgrade_nodes_repos_6_to_7
-        onadmin upgrade_ses_to_4
+        cloudsource=$upgrade_cloudsource onadmin prepare_cloudupgrade_nodes_repos
+        if [[ $cloudsource = develcloud6 ]]; then
```
Comment (bmwiedemann) [inline scripts/mkcloud]: Should this be `if iscloudver 6 ; then` or would there be a reason to not upgrade SES for `cloudsource=GM6+up` ?
Could also be written as
```bash
iscloudver 6 && onadmin upgrade_ses_to_4
```

File: scripts/qa_crowbarsetup.sh [hunk 4]
```diff
@@ -4675,9 +4675,16 @@ function onadmin_prepare_cloudupgrade_admin_repos_6_to_7
...
 
-    # change system repositories to SP2
-    $zypper rr SLES12-SP1-Pool
-    $zypper rr SLES12-SP1-Updates
+    # remove old repositories
+    # (so that zypper does not have an option to choose from the old ones)
+    if [[ $cloudsource = develcloud7 ]]; then
+        $zypper rr SLES12-SP1-Pool
+        $zypper rr SLES12-SP1-Updates
+    elif [[ $cloudsource = develcloud8 ]]; then
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: IMHO `iscloudver` would be better here and above

File: scripts/mkcloud (hunk 3 above)
Comment (jsuchome) [reply scripts/mkcloud]: I like `iscloudver 6 && onadmin upgrade_ses_to_4`

**Comment by jdsn** [comment]
looks good so far - waiting for the WIP label to be removed for a final review

**Pull request merged**